File: Submissions/#0369_div2/A/main.java
import java.util.*;

public class Main {

  public static void main (String[] args) {

    Scanner sc = new Scanner(System.in);
    int n = sc.nextInt();
    
    String[][] v = new String[n][5];
    String answer = "NO";
    for (int i = 0; i < n; i++) {
      String V = sc.next();
      v[i] = V.split("");
      if (answer.equals("NO")) {
        if (v[i][0].equals("O") && v[i][1].equals("O")) {
          answer = "YES";
          v[i][0] = "+";
          v[i][1] = "+";
        } else if (v[i][3].equals("O") && v[i][4].equals("O")){
          answer = "YES";
          v[i][3] = "+";
          v[i][4] = "+";
        }
      }
    }
    
    sc.close();

    System.out.println(answer);
    if (answer.equals("YES")) {
      for (int i = 0; i < n; i++) {
        for (int j = 0; j < 5; j++) {
          System.out.print(v[i][j]);
        }
        System.out.println("");
      }
    }

  }

}

File: Submissions/#0369_div2/A/main.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {

  int n;
  cin >> n;

  vector<string> v(n);
  string answer = "NO";
  for (int i = 0; i < n; i++) {
    cin >> v[i];
    if (answer == "NO") {
      if (v[i][0] == 'O' && v[i][1] == 'O') {
        answer = "YES";
        v[i][0] = '+';
        v[i][1] = '+';
      } else if (v[i][3] == 'O' && v[i][4] == 'O'){
        answer = "YES";
        v[i][3] = '+';
        v[i][4] = '+';
      }
    }
  }
  
  cout << answer << endl;
  if (answer == "YES") {
    for (int i = 0; i < n; i++) {
      cout << v[i] << endl;
    }
  }

}